itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java

```diff
@@ -0,0 +1,350 @@
+        .run("show tables like 't2'")
+        .verifyResults(new String[] { "t2" })
+        .run("select place from t2")
+        .verifyResults(new String[] { "bangalore" });
+
+    // add new  data externally, to a partition, but under the table level top directory
+    Path partitionDir = new Path(externalTableLocation, "country=india");
+    try (FSDataOutputStream outputStream = fs.create(new Path(partitionDir, "file.txt"))) {
+      outputStream.write("pune\n".getBytes());
+      outputStream.write("mumbai\n".getBytes());
+    }
+
+    tuple = primary.run("use " + primaryDbName)
+        .run("insert into t2 partition(country='australia') values ('sydney')")
+        .dump(primaryDbName, tuple.lastReplicationId);
+
+    assertExternalFileInfo(Collections.singletonList("t2"),
+        new Path(tuple.dumpLocation, FILE_NAME));
+
+    replica.load(replicatedDbName, tuple.dumpLocation, loadWithClause)
+        .run("use " + replicatedDbName)
+        .run("select distinct(country) from t2")
+        .verifyResults(new String[] { "india", "australia" })
+        .run("select place from t2 where country='india'")
+        .verifyResults(new String[] { "bangalore", "pune", "mumbai" })
+        .run("select place from t2 where country='australia'")
+        .verifyResults(new String[] { "sydney" });
+
+    Path customPartitionLocation =
+        new Path("/" + testName.getMethodName() + "/partition_data/t2/country=france");
+    fs.mkdirs(externalTableLocation, new FsPermission("777"));
+
+    // add new partitions to the table, at an external location than the table level directory
+    try (FSDataOutputStream outputStream = fs
+        .create(new Path(customPartitionLocation, "file.txt"))) {
+      outputStream.write("paris".getBytes());
+    }
+
+    tuple = primary.run("use " + primaryDbName)
+        .run("ALTER TABLE t2 ADD PARTITION (country='france') LOCATION '" + customPartitionLocation
+            .toString() + "'")
+        .dump(primaryDbName, tuple.lastReplicationId);
+
+    replica.load(replicatedDbName, tuple.dumpLocation, loadWithClause)
+        .run("use " + replicatedDbName)
+        .run("select place from t2 where country='france'")
+        .verifyResults(new String[] { "paris" });
+
+    // change the location of the partition via alter command
+    String tmpLocation = "/tmp/" + System.nanoTime();
+    primary.miniDFSCluster.getFileSystem().mkdirs(new Path(tmpLocation), new FsPermission("777"));
+
+    tuple = primary.run("use " + primaryDbName)
+        .run("alter table t2 partition (country='france') set location '" + tmpLocation + "'")
+        .dump(primaryDbName, tuple.lastReplicationId);
+
+    replica.load(replicatedDbName, tuple.dumpLocation, loadWithClause)
+        .run("use " + replicatedDbName)
+        .run("select place from t2 where country='france'")
+        .verifyResults(new String[] {});
+  }
+
+  @Test
+  public void externalTableIncrementalReplication() throws Throwable {
```

Comment:
we have separate ones for incremental and bootstrap so that test dont become too long and those phases go through different code paths as well. Agree that a test case for drop makes sense will add that.